### PR TITLE
feat: Multichain deployments + Pragma Oracle deployment flow

### LINKED ISSUE #51: dev: support multichain deployment
Add support for multichain deployment with subfolders.

### PR DESCRIPTION
**Pull request by akhercha**
Resolves: #51

## Work done

* Multi chain deployments
* Pragma Oracle

Todo:
* [x] Fix initiliazer of Pragma.sol
* [x] Fix initializer of PragmaDispatcher
* [x] Add --deterministic or similar params for deterministic deployment addresses

### CODE REVIEW

File: config/currencies.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+- name: "Bitcoin"
```
Comment (EvolveArt) [inline config/currencies.yaml:1]: let's just fetch it remotely from the sdk repo no?

Comment (akhercha) [reply config/currencies.yaml:1]: Mmmh, not worth I'd say, no need to create a dependency between those repos, we'll see when we have a proper source of truth? (the config doesn't change a lot / isnt too big anyway)

Comment (EvolveArt) [reply config/currencies.yaml:1]: but in any case if its not registered in the sdk config it's not going to be published assuming everyone uses the SDK to publish

Comment (akhercha) [reply config/currencies.yaml:1]: should probably be the other way around then - pragma-sdk importing the config here

**Pull request merged**